tests/nodejs_install.etest

```diff
@@ -0,0 +1,350 @@
+        return 0
+    }'
+
+    emock "pkg_install" '
+    {
+        return 0
+    }'
+
+    # Make the call
+    install_nodejs --install-prefix "${test_prefix_dir}" --lts-codename "${lts_codename}"
+
+    # NOTE: This test's intent is to verify the proper calls, since we don't want to duplicate the
+    #       testing that already occurs for the individual ebash built-ins
+    # Verify the mocks being called correctly
+    assert_emock_called_with "get_latest_node_version" 0 \
+        --lts-codename "${lts_codename}"                 \
+        --sha-output-path "${default_sha_path}"
+
+    assert_emock_called_with "os" 0 \
+        darwin
+
+    assert_emock_called_with "pkg_install" 0 \
+        --sync "${expected_package}"
+}
+
+ETEST_install_prefix_real_files()
+{
+    # Establish a test directory to install into
+    local test_prefix
+    test_prefix="$(mktemp -u usr_local.XXXXXX)"
+    local test_prefix_dir="${TEST_DIR_OUTPUT}/${test_prefix}"
+    local test_file="${test_prefix_dir}/bin/node"
+
+    # Simulate existing install creating a real file, not a symlink
+    mkdir -p "${test_prefix_dir}/bin"
+    echo "I think therefor I am" > "${test_file}"
+
+    # Verify the call fails
+    assert_false cleanup_node_symlinks --install-prefix "${test_prefix_dir}"
+
+    # Make sure the file wasn't removed
+    assert_exists "${test_file}"
+}
+
+ETEST_node_installation_check()
+{
+    emock "which" '
+    {
+        if [[ arg[0] == "node" ]]; then
+            echo "/usr/local/bin/node"
+            return 0
+        else
+            return 1
+        fi
+    }'
+    check_for_node_installation || assert_eq 3 "${?}"
+}
+
+ETEST_npm_installation_check()
+{
+    emock "which" '
+    {
+        if [[ arg[0] == "npm" ]]; then
+            echo "/usr/local/bin/npm"
+            return 0
+        else
+            return 1
+        fi
+    }'
+    check_for_node_installation || assert_eq 2 "${?}"
+}
+
+ETEST_nvm_installation_check()
+{
+    emock "which" '
+    {
+        if [[ arg[0] == "nvm" ]]; then
+            echo "/usr/local/bin/nvm"
+            return 0
+        else
+            return 1
+        fi
+    }'
+    check_for_node_installation || assert_eq 1 "${?}"
+}
+
+ETEST_get_latest_node_version()
+{
+    local codename="gallium"
+    local expected_version="16.13.2"
+    local sha_file="${TEST_DIR_OUTPUT}/nodejs_SHASUMS256.txt"
+
+    # Establish mock
+    emock "retrieve_sha" '
+    {
+        return 0
+    }'
+
+    # Test Legit version
+    cat > "${sha_file}" <<-END
+451367428b40095656133  node-v16.13.2-aix-ppc64.tar.gz
+END
+    actual_version=$(get_latest_node_version --lts-codename "${codename}" --sha-output-path "${sha_file}")
+    assert_eq "${expected_version}" "${actual_version}"
+
+    # Test Bogus version
+    cat > "${sha_file}" <<-END
+451367428b40095656133  node-.13.2-aix-ppc64.tar.gz
+END
+
+    try
+    {
+        actual_version=$(get_latest_node_version --lts-codename "${codename}" --sha-output-path "${sha_file}")
```

Comment:
Can this be change to ` assert_false get_latest_node_version --lts-codename "${codename}" --sha-output-path "${sha_file}" ?` without using try and catch?

Reply:
Yep! I thought I needed it in a subshell (which would have required it), but I don't! Fixed and much cleaner!

Reply:
Thanks for updating all the tests!